includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2155,43 +2133,19 @@ public function add_twentytwentyone_dark_mode_toggle() {
 	 * Make the mobile menu for the Twenty Twenty-One theme AMP compatible.
 	 */
 	public function add_twentytwentyone_mobile_modal() {
-		$menu_query  = $this->dom->xpath->query( "//div[ @class and contains( concat( ' ', normalize-space( @class ), ' ' ), ' primary-menu-container ' ) ]" );
 		$menu_toggle = $this->dom->getElementById( 'primary-mobile-menu' );
 
-		if ( 1 !== $menu_query->length || ! $menu_toggle ) {
+		if ( ! $menu_toggle ) {
 			return;
 		}
 
-		/** @var DOMElement $primary_menu */
-		$primary_menu      = $menu_query->item( 0 );
-		$primary_menu_copy = $primary_menu->cloneNode( true );
-		foreach ( $this->dom->xpath->query( './/*[ @id ]', $primary_menu_copy ) as $element ) {
-			/** @var DOMElement $element */
-			$element->setAttribute( 'id', $element->getAttribute( 'id' ) . '-mobile' );
-		}
-
-		$body_id = $this->dom->getElementId( $this->dom->body, 'body' );
-
-		// Create an <amp-lightbox> element that will contain the modal.
-		$amp_lightbox = $this->dom->createElement( 'amp-lightbox' );
-		$amp_lightbox->setAttribute( 'layout', 'nodisplay' );
-		$amp_lightbox->setAttribute( 'animate-in', 'fade-in' );
-		$amp_lightbox->setAttribute( 'scrollable', true );
-		$amp_lightbox_id = $this->dom->getElementId( $amp_lightbox );
-
-		$state_string = str_replace( '-', '_', $amp_lightbox_id );
-
-		AMP_DOM_Utils::add_amp_action( $menu_toggle, 'tap', "{$amp_lightbox_id}.open" );
-		AMP_DOM_Utils::add_amp_action( $menu_toggle, 'tap', "{$body_id}.toggleClass(class=primary-navigation-open,force=true)" );
-
-		AMP_DOM_Utils::add_amp_action( $amp_lightbox, 'lightboxOpen', "AMP.setState({{$state_string}:true})" );
-		AMP_DOM_Utils::add_amp_action( $amp_lightbox, 'lightboxClose', "AMP.setState({{$state_string}:false})" );
-		AMP_DOM_Utils::add_amp_action( $amp_lightbox, 'lightboxClose', "{$body_id}.toggleClass(class=primary-navigation-open,force=false)" );
+		$state_string = 'mobile_menu_toggled';
+		$body_id      = $this->dom->getElementId( $this->dom->body, 'body' );
 
+		AMP_DOM_Utils::add_amp_action( $menu_toggle, 'tap', "AMP.setState({{$state_string}: !{$state_string}})" );
```

Comment:
The disadvantage with this approach is that the mobile menu can no longer be closed when the <key>Esc</key> key, is pressed, but as @westonruter pointed out it's unlikely that a user would be using a keyboard to navigate on mobile.